### PR TITLE
Salesforce contact company

### PR DESCRIPTION
**Pull request by mqueme**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/4853
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
When syncing Contacts from Salseforce to Mautic we were not able to sync companies, because we were not pulling reference fields from Mautic, this PR adds a exception for AccountId related to a contact, to be able to map it to the company field in the Mautic Contact.

#### Steps to test this PR:
1. Map Contact Company (Account ID) in the contacts tab from Salesforce fields to the company field in Mautic. This should map the contact created/updated in Mautic to a Mautic company.

### CODE REVIEW

**Comment by virlatinus** [comment]
@mqueme I'm getting this error while executing `fetchleads` in dev:
```
[Doctrine\DBAL\Exception\DriverException]                                                                                        
  An exception occurred while executing 'SELECT DISTINCT(i.integration_entity_id), i.integration_entity, i.id, i.internal_entity_  
  id,l.email, l.lastname, l.company, l.firstname FROM integration_entity i INNER JOIN leads l ON l.id = i.internal_entity_id WHER  
  E (i.integration = ?) AND (i.integration_entity = ?) AND (i.internal_entity = ?) AND ((i.integration_entity_id IS NOT NULL) AND  
   (((i.last_sync_date IS NOT NULL) AND (l.date_modified > i.last_sync_date)) OR ((i.last_sync_date IS NULL) AND (l.date_modified  
   IS NOT NULL) AND (l.date_modified > l.date_added)))) AND (l.email IS NOT NULL) AND (l.date_modified BETWEEN ? and ?) GROUP BY   
  i.integration_entity, i.integration_entity_id LIMIT 100' with params ["Salesforce", "Contact", "lead", "2017-10-03 02:29:36", "  
  2017-10-10 02:29:36"]:                                                                                                           
  SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #3 of SELECT list is not in GROUP BY clause and contains non  
  aggregated column 'mautic.i.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sq  
  l_mode=only_full_group_by         
```

**Comment by Maxell92** [comment]
Tried and got this error:
screen shot 2017-10-10 at 23 25 22

File: plugins/MauticCrmBundle/Integration/SalesforceIntegration.php [hunk 1]
```diff
@@ -1751,7 +1762,7 @@ protected function getMauticContactsToUpdate(
             if (!empty($lead['email'])) {
                 $lead                                               = $this->getCompoundMauticFields($lead);
                 $key                                                = $this->getSyncKey($lead['email']);
-                $trackedContacts[$lead['integration_entity']][$key] = $lead['id'];
+                $trackedContacts[$lead['integration_entity']][$key] = $lead['internal_entity_id'];
```
Comment (alanhartless) [inline plugins/MauticCrmBundle/Integration/SalesforceIntegration.php]: Why was this changed? It should be the ID of the IntegrationEntity as it's used to obtain references to it in other places. https://github.com/mautic/mautic/blob/3fc67864148828ec4c2f78676be149a276e8e5ac/plugins/MauticCrmBundle/Integration/SalesforceIntegration.php#L2321

File: app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php [hunk 2]
```diff
@@ -189,13 +189,13 @@ public function findLeadsToUpdate(
...
         if (false === $limit) {
-            $q->select('count(DISTINCT(i.integration_entity_id)) as total');
+            $q->select('count(i.integration_entity_id) as total');
 
             if ($integrationEntity) {
                 $q->addSelect('i.integration_entity');
             }
         } else {
-            $q->select('DISTINCT(i.integration_entity_id), i.integration_entity, i.id, i.internal_entity_id,'.$leadFields);
+            $q->select('DISTINCT i.integration_entity_id, i.integration_entity, i.internal_entity_id,'.$leadFields);
```
Comment (alanhartless) [inline app/bundles/PluginBundle/Entity/IntegrationEntityRepository.php]: i.id is required for $trackedContacts below

**Comment by alanhartless** [comment]
On the error Petr obtained, looks like we need to escape single quotes when communicating company names to SF.

**Comment by javjim-mautic** [comment]
Tested and works properly
